src/auth/integration-tests/tests/driver.rs

```diff
@@ -76,4 +80,44 @@ mod driver {
 
         Ok(())
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn api_key() -> Result<()> {
+        let project = std::env::var("GOOGLE_CLOUD_PROJECT").expect("GOOGLE_CLOUD_PROJECT not set");
+
+        // Create a SecretManager client. When running on GCB, this loads MDS
+        // credentials for our `integration-test-runner` service account.
+        let client = SecretManagerService::new().await?;
+
+        // Load the API key under test.
+        let response = client
+            .access_secret_version(format!(
+                "projects/{}/secrets/test-api-key/versions/latest",
+                project
+            ))
+            .send()
+            .await?;
+        let api_key = response
+            .payload
+            .expect("missing payload in test-api-key response")
+            .data;
```

Comment:
FWIW, you can expose secrets to GCB:

https://cloud.google.com/build/docs/securing-builds/use-secrets